src/main/java/org/eclipse/basyx/extensions/aas/aggregator/aasxupload/AASAggregatorAASXUpload.java

```diff
@@ -65,11 +75,47 @@ public void uploadAASX(InputStream aasxStream) {
 			AASXToMetamodelConverter converter = new AASXToMetamodelConverter(aasxStream);
 			Set<AASBundle> bundles = converter.retrieveAASBundles();
 			AASBundleHelper.integrate(this, bundles);
+			uploadFilesInAASX(converter);
 		} catch (Exception e) {
 			throw new MalformedRequestException("invalid request to aasx path without valid aasx input stream");
 		}
 	}
 
+	public void uploadFilesInAASX(AASXToMetamodelConverter converter) throws InvalidFormatException, ParserConfigurationException, SAXException, IOException {
+		converter.retrieveAASBundles().forEach(aasBundle -> {
+			aasBundle.getSubmodels().forEach(submodel -> {
+				submodel.getSubmodelElements().values().forEach(sme -> {
+					uploadFileInSubmodelElement(aasBundle, converter, submodel, sme, getSubmodelElementFileUploadURL(submodel.getIdShort(), sme.getIdShort()));
+
+					if (sme instanceof SubmodelElementCollection) {
```

Comment:
This line will only check it once if there is a SMC there.

&rarr; you will need to make this a recursive function. I suggest moving lines 88-95 into an extra function and calling that in line 93 instead of the uploadFileInSubmodelElement.

````java

	public void uploadFilesInAASX(AASXToMetamodelConverter converter) throws InvalidFormatException, ParserConfigurationException, SAXException, IOException {
		converter.retrieveAASBundles().forEach(aasBundle -> {
			aasBundle.getSubmodels().forEach(submodel -> {
				submodel.getSubmodelElements().values().forEach(sme -> {
					uploadWithRecurringSMC(aasBundle, converter, submodel, sme);
				});
			});
		});
	}
	
	private void uploadWithRecurringSMC(AASBundle aasBundle, AASXToMetamodelConverter converter, ISubmodel submodel, ISubmodelElement sme) {
		uploadFileInSubmodelElement(aasBundle, converter, submodel, sme, getSubmodelElementFileUploadURL(submodel.getIdShort(), sme.getIdShort()));

		if (sme instanceof SubmodelElementCollection) {
			ISubmodelElementCollection smeCollection = (ISubmodelElementCollection) sme;
			smeCollection.getSubmodelElements().values().forEach(element -> {
				uploadWithRecurringSMC(aasBundle, converter, submodel, sme);
			});
		}
	}
````

Something like this. Maybe rework the method signature, as it's just an example.